mainwindow.cpp

```diff
@@ -1124,10 +1104,18 @@ void MainWindow::updateHashTypes(const QStringList &pathToPwdFile, const QString
         // isn't too late, otherwise we don't replace the model
         model->fillHashTypes(detailedTypesPerRow);
         m_ui->tableView_Hashes->setModel(model);
+        QString savedFormat = m_ui->comboBox_Format->currentText();
         // For jumbo, we list only available formats in file in attack option
         m_ui->comboBox_Format->clear();
         m_ui->comboBox_Format->addItem(tr("Auto detect"));
         m_ui->comboBox_Format->addItems(listOfTypesInFile);
+        // Restore user's selection
+        int indexSavedFormat = m_ui->comboBox_Format->findText(savedFormat);
+        if (indexSavedFormat != -1) {
+            m_ui->comboBox_Format->setCurrentIndex(indexSavedFormat);
+        } else if(savedFormat.isEmpty()){
+            m_ui->comboBox_Format->setEditText(savedFormat);
```

Comment:
You don't need to do this now, but I think you're aware by now that we should gradually remove the _ and - chars from object names and method names. They're reminiscent of C code and are unnecessary with proper and concise naming. The only exception are the member vars prefixed with m_ of course. :-)